services/src/main/java/io/grpc/protobuf/services/ProtoReflectionService.java

```diff
@@ -68,11 +68,8 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
 public final class ProtoReflectionService extends ServerReflectionGrpc.ServerReflectionImplBase {
 
-  private final Object lock = new Object();
-
+  private final Map<Server, ServerReflectionIndex> serverReflectionIndexes = new WeakHashMap<>();
```

Comment:
`@GuardedBy("this")`?

Reply:
Does `@GuardedBy` checks accessing collection elements? I thought I was just the field itself.

Reply:
It would notice most method invocations on the map, like get/put.

Reply:
Oh, good to know. Thanks. Added back.